### ISSUE TITLE
SA1202 and Read Only Dependency Properties

### ISSUE THREAD

**Issue by osttocs** [DotNetAnalyzers/StyleCopAnalyzers#2548]
Rule SA1202 is being reported for the read only dependency property pattern in WPF where a static, private DependencyPropertyKey is declared before a static, public DependencyProperty.  This issues was resolved in the previous edition of StyleCop by https://github.com/StyleCop/StyleCop/issues/51.

**Comment by gtbuchanan** [comment]
I can confirm this is still an issue, though I'm using Xamarin rather than WPF. Using a static constructor is a workaround unless you enable [CA1810](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1810). We're currently just suppressing SA1202 for these situations, but it is rather verbose.

**Comment by sharwell** [comment]
@gtbuchanan Can you show an example for how `BindableProperty` is used in source?

**Comment by gtbuchanan** [comment]
@sharwell It's very similar to WPF. It [looks to be the same](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/bindable-properties?view=net-maui-8.0) in MAUI as well.
```csharp
private static readonly BindablePropertyKey IsLoadedPropertyKey = BindableProperty
    .CreateReadOnly(nameof(IsLoaded), typeof(bool), typeof(BaseLazyView), default);

/// <summary>
/// This is a read-only <see cref="BindableProperty"/> that indicates when the view is loaded.
/// </summary>
public static readonly BindableProperty IsLoadedProperty = IsLoadedPropertyKey.BindableProperty;

/// <summary>
/// Gets or sets a value indicating whether the view is loaded.
/// </summary>
public bool IsLoaded
{
    get => (bool)this.GetValue(IsLoadedProperty);
    set => this.SetValue(IsLoadedPropertyKey, value);
}
```

**Comment by sharwell** [comment]
What happens if `IsLoadedProperty` is a property?

```
public static BindableProperty IsLoadedProperty => IsLoadedPropertyKey.BindableProperty;
```

**Comment by gtbuchanan** [comment]
Honestly, I haven't tried because the [documentation](https://learn.microsoft.com/en-us/xamarin/xamarin-forms/xaml/bindable-properties) states explicitly that the `BindableProperty` should be a `public static readonly` field. I'll give it a go and report back later.

**Comment by gtbuchanan** [comment]
I tested in Xamarin Forms with an auto-implemented get-only property as you suggested and it seems to work fine as far as XAML binding is concerned. I couldn't find any documentation about why a `public static readonly` field is required. Perhaps it is just a strong suggestion for performance reasons.
